### PR TITLE
Add axios readme with information on how to download a file

### PR DESCRIPTION
**Pull request by ShockwaVee**
Axios in JS

### CODE REVIEW

**Comment by ShockwaVee** [comment]
The reason why this is called Axios is because of the exact reason you described. I had no idea it was used outside of Axios :D
Do you suggest I rename this segment to something like file-downloads or api-communication? File downloads would cover a much narrower topic, right of the bat.

File: playbooks/javascript/axios.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+There are two ways to allow a user to download a file from your API:
+- using a direct link to the resource and using the "download" tag
+- fetching a file and initiating the download sequence
+
+### Direct link
+This option is pretty easy to implement and is not connected to Axios all that much. Basically, after you fetch a resource link from your API, you can use an anchor tag and set the resource link as the href. After you've done that, all you have to do is set a download tag which. This will make the browser initiate the download modal and allow the user to download a file. This will also work with local files. You can see more about this method [here](https://www.w3schools.com/tags/att_a_download.asp).
+
+### Fetching a file
+In case the file is being generated on the fly and isn't saved anywhere, there are a couple of things to watch out for when downloading such files.
+1. Make sure your [request config](https://www.w3schools.com/tags/att_a_download.asp) has `responseType` has the value `blob` instead of `json`. This will indicate the type of data the server will respond with.
```
[the remark this answers is not in the document]
Comment (ShockwaVee) [reply playbooks/javascript/axios.md:16]: You're right, I'll fix it right away.

**Comment by renatoruk** [comment]
> The reason why this is called Axios is because of the exact reason you described. I had no idea it was used outside of Axios :D
> Do you suggest I rename this segment to something like file-downloads or api-communication? File downloads would cover a much narrower topic, right of the bat.

Haha, I have a similar experience of getting to know this API.
I must say that I learned about `responseType` at the same time while developing a similar file downloads feature, and because that project used fetch API, I needed to dig deeper on the natives. That's what led me into thinking that the responseType from axios is not something that's axios only.

I like the name you suggested, file-downloads, it's very to the point

**Comment by renatoruk** [comment]
Awesome, merging!

**Pull request merged**